### PR TITLE
feat(jtk): add --name filter to fields list

### LINKED ISSUE #194: feat(jtk): add --name filter to fields list
## Problem

AI agents frequently run `jtk fields list --project MON | grep -i "point|story"` trying to find the story points field ID. The `--project` flag doesn't exist, Cobra errors, grep finds nothing, and the agent gives up — even though the field exists in the global list.

## Solution

Add a `--name` flag to `jtk fields list` for case-insensitive substring filtering on field names. Client-side filtering of the already-fetched global field list — no API changes needed.

```bash
jtk fields list --name "story point"
jtk fields list --name "story" --custom -o json
```

## Scope

- Add `--name` flag and filtering logic to `fields.go`
- Unit tests for filtering, case insensitivity, composition with `--custom`
- Update README command docs
- Add integration test cases

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- Adds `--name` flag to `jtk fields list` for case-insensitive substring filtering on field names
- AI agents frequently pass `--project` (which doesn't exist), causing Cobra errors — `--name` gives a direct way to find fields like `jtk fields list --name "story point"`
- Client-side filtering of the global field list, composes with `--custom`
- Updated README command docs and integration test cases

Closes #194

## Test plan

- [x] 5 new unit tests: name filter, case insensitivity, no match, JSON output, composition with `--custom`
- [x] Updated 3 existing `runList` test calls for new parameter
- [x] `make build` passes
- [x] `make test` passes
- [x] Integration test cases added to `integration-tests.md`

### CODE REVIEW

**Comment by rianjs** [comment]
## TDD Assessment

**PR:** Add `--name` filter to `jtk fields list`

### Coverage: Sufficient

The logic is simple (client-side substring filter after fetching) and the test suite covers the meaningful cases well.

**What's covered:**
- Basic name filtering (matches included, non-matches excluded)
- Case-insensitive matching (`STORY` matches `Story Points`)
- No-match path returns `"No fields found"` message
- JSON output with filter applied
- Combined `--name` + `--custom` flags
- Flag registration with correct default value
- Existing tests updated to pass the new `nameFilter` param

**Minor gaps (not blocking):**
- No test for empty string `nameFilter` behavior (i.e., `""` skips filtering). This is implicit in the existing `TestRunList_Table`, `TestRunList_JSON`, and `TestRunList_Empty` tests since they now pass `""`, but there's no explicit assertion that an empty filter returns all fields. Low risk given the guard is `if nameFilter != ""`.
- No test for whitespace-only filter (e.g., `"   "`) — would currently pass through to `strings.Contains` and match nothing, silently returning "No fields found" rather than treating it as an unset filter. Edge case, not a real-world concern.
- `TestRunList_NameFilter_WithCustom` comment says "GetCustomFields returns only custom fields" but the mock server doesn't differentiate on the endpoint — the `customOnly` flag routes to a different API call (`GetCustomFields` vs `GetAllFields`). The test effectively validates that name filtering works after the custom-only path, which is the meaningful thing to test. Comment is slightly misleading but test logic is correct.

**Verdict:** Approve on coverage. The new logic is straightforward and the added tests are proportionate to the change.

File: tools/jtk/internal/cmd/fields/fields_test.go [hunk 1]
```diff
@@ -98,11 +102,127 @@ func TestRunList_Empty(t *testing.T) {
...
+	opts := &root.Options{Output: "json", Stdout: &stdout, Stderr: &bytes.Buffer{}}
+	opts.SetAPIClient(client)
+
+	err = runList(context.Background(), opts, false, "story")
+	testutil.RequireNoError(t, err)
+	testutil.Contains(t, stdout.String(), "Story Points")
+	testutil.NotContains(t, stdout.String(), "Summary")
+}
+
+func TestRunList_NameFilter_WithCustom(t *testing.T) {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/fields/fields_test.go:202]: 🔵 **Low** (harness-engineering:architecture-reviewer): TestRunList_NameFilter_WithCustom doesn't verify that `GetCustomFields` (vs `GetFields`) is actually called when `customOnly=true`. The mock server returns the same response regardless of which endpoint is hit, so a bug in the routing logic (lines 74-78 of fields.go) wouldn't be caught. Consider checking `r.URL.Path` in the mock handler to confirm the correct API endpoint is called. The routing logic is trivially correct today, so this is a test quality suggestion rather than a blocking issue.

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `3bb5c97`

> **Approved** with 1 non-blocking suggestion below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:architecture-reviewer | 1 |

<details>
<summary><strong>harness-engineering:architecture-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/fields/fields_test.go:202`
> TestRunList_NameFilter_WithCustom doesn't verify that `GetCustomFields` (vs `GetFields`) is actually called when `customOnly=true`. The mock server returns the same response regardless of which endpoint is hit, so a bug in the routing logic (lines 74-78 of fields.go) wouldn't be caught. Consider checking `r.URL.Path` in the mock handler to confirm the correct API endpoint is called. The routing logic is trivially correct today, so this is a test quality suggestion rather than a blocking issue.

</details>

---
<details>
<summary>Completed in 2m 22s | $0.56</summary>

| Field | Value |
|---|---|
| Reviewers | [hybrid-synthesis](https://github.com/SignalFT/pr-review-daemon/blob/main/pr_review_daemon/reviewers/prompt.py), [database:sql-reviewer](https://github.com/SignalFT/monit-claude-plugins/tree/main/plugins/database/agents/sql-reviewer), [security:code-auditor](https://github.com/SignalFT/monit-claude-plugins/tree/main/plugins/security/agents/code-auditor), [harness-engineering:knowledge-reviewer](https://github.com/SignalFT/monit-claude-plugins/tree/main/plugins/harness-engineering/agents/knowledge-reviewer), [harness-engineering:enforcement-reviewer](https://github.com/SignalFT/monit-claude-plugins/tree/main/plugins/harness-engineering/agents/enforcement-reviewer), [harness-engineering:architecture-reviewer](https://github.com/SignalFT/monit-claude-plugins/tree/main/plugins/harness-engineering/agents/architecture-reviewer), [harness-engineering:legibility-reviewer](https://github.com/SignalFT/monit-claude-plugins/tree/main/plugins/harness-engineering/agents/legibility-reviewer) |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 22s (Reviewers: 22s · Synthesis: 47s) |
| Cost | $0.56 |
| Tokens | 143.6k in / 7.9k out |
| Turns | 9 |

</details>
